glean-core/ios/Glean/Metrics/Ping.swift

```diff
@@ -4,22 +4,55 @@
 
 import Foundation
 
+public protocol ReasonCodes: Hashable {
+    /// The index of the extra key, used to index the string array passed at `EventMetricType` instantiation.
+    func index() -> Int
+}
+
+/// Default of no extra keys for events.
+///
+/// An enum with no values for convenient use as the default set of extra keys
+/// that an `EventMetricType` can accept.
+public enum NoReasonCodes: ReasonCodes {
+    public func index() -> Int {
+        return 0
+    }
+}
+
+public class PingBase {
```

Comment:
I think changing this to `PingBase` may be why the integration tests are failing.  It may take a tweak to the Xcode build to update the include to make this work. (Sorry, it's not ideal, I know).  Let me take a deeper look tomorrow and see if I can help.

Reply:
Thanks.  No rush on this!

Reply:
I took another look at this and the only thing I can think of is that Carthage is not getting the iOS version of Glean with the changes.  I'm a bit stumped as to why this isn't working and I'm trying to set up a build locally that uses a locally built Glean without Carthage to see if it still fails to build.  @badboy do you have any ideas?

Reply:
We might just be generating invalid Swift code with glean_parser.
Last time I looked at it it wasn't yet fully updated to accomodate the reason codes.

I would need to look into the generated code to determine what's happening.

Reply:
glean_parser should be generating valid code now -- it works in the tests of Glean iOS proper, and we are already using this feature there.

Reply:
In that case I need to take a closer look

Reply:
It seems to be fixed.